build/vanagon/components/openssl.rb

```diff
@@ -0,0 +1,98 @@
+component 'openssl' do |pkg, settings, platform|
+  pkg.version '1.1.1d'
+  pkg.md5sum '3be209000dbc7e1b95bcdf47980a3baa'
+  pkg.url "https://openssl.org/source/openssl-#{pkg.get_version}.tar.gz"
+
+  #############################
+  # ENVIRONMENT, FLAGS, TARGETS
+  #############################
+
+  target = cflags = ldflags = sslflags = ''
+
+  pkg.environment 'PATH', "/opt/pl-build-tools/bin:$$PATH"
+  pkg.environment 'CC', "/opt/pl-build-tools/bin/#{settings[:platform_triple]}-gcc"
+
+  cflags = "#{settings[:cflags]} -fPIC"
+
+  ldflags = "-Wl,-rpath=/opt/pl-build-tools/#{settings[:platform_triple]}/lib -Wl,-rpath=#{settings[:libdir]} -L/opt/pl-build-tools/#{settings[:platform_triple]}/lib"
+  pkg.environment 'PATH', '/opt/pl-build-tools/bin:$(PATH):/usr/local/bin'
+
+  cflags = settings[:cflags]
+  ldflags = "#{settings[:ldflags]} -Wl,-z,relro"
+  if platform.architecture =~ /86$/
+    target = 'linux-elf'
+    sslflags = '386'
+  elsif platform.architecture =~ /64$/
+    target = 'linux-x86_64'
+  end
+
+  ####################
+  # BUILD REQUIREMENTS
+  ####################
+
+  pkg.build_requires "runtime-#{settings[:runtime_project]}"
+
+  ###########
+  # CONFIGURE
+  ###########
+
+#  if platform.is_cross_compiled?
+#    pkg.apply_patch 'resources/patches/openssl/openssl-1.1.1a-revert-7a061312.patch'
+#  end
+
+  # OpenSSL Configure doesn't honor CFLAGS or LDFLAGS as environment variables.
+  # Instead, those should be passed to Configure at the end of its options, as
+  # any unrecognized options are passed straight through to ${CC}. Defining
+  # --libdir ensures that we avoid the multilib (lib/ vs. lib64/) problem,
+  # since configure uses the existence of a lib64 directory to determine
+  # if it should install its own libs into a multilib dir. Yay OpenSSL!
+  configure_flags = [
```

Comment:
Arch linux also sets enable-ec_nistp_64_gcc_128
might be useful here too
source: https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/openssl